python/ml4ir/base/model/architectures/auto_dag_network.py

```diff
@@ -239,15 +274,24 @@ def __init__(
 
         # Get all available layers (including keras layers)
         self.available_layers = get_layer_subclasses()
-        model_graph = self.define_architecture(model_config)
-        self.execution_order: List[LayerNode] = model_graph.topological_sort()
+        self.model_graph = self.define_architecture(model_config)
+
+        # Get the first file handler for the logger
+        file_handlers = [handler for handler in self.file_io.logger.handlers if isinstance(handler, FileHandler)]
+        logging_dir = Path(file_handlers[0].baseFilename).parent if file_handlers else Path(self.DEFAULT_VIZ_SAVE_PATH)
```

Comment:
@arvindsrikantan This seems unnecessarily complicated. I recommend you move it to `RelevanceScorer` perhaps or even `RelevanceModel`?

Reply:
Move the entire visualization code to those classes you mean? It's very specific to AutoDagNework though